.github/workflows/test_dbt_models.yaml

```diff
@@ -23,37 +23,13 @@ jobs:
           role-to-assume: ${{ secrets.AWS_IAM_ROLE_TO_ASSUME_ARN }}
 
       - name: Install Python requirements
-        run: pip install -r scripts/requirements.transform_dbt_test_results.txt
+        run: pip install -r scripts/requirements.run_iasworld_data_tests.txt
         working-directory: ${{ env.PROJECT_DIR }}
         shell: bash
 
       - name: Run tests
         run: |
-          # dbt doesn't differentiate between test failures and errors, but we
-          # need to since we expect failures. Do this by capturing the output
-          # and checking its contents to look for errors
-          if output=$(dbt test --target "$TARGET" --selector qc_tests --store-failures); then
-            echo "$output"
-          else
-            status_code="$?"
-            if [[ "$output" =~ "Runtime Error" || "$output" =~ "Compilation Error" ]]; then
-              # The presence of an error string indicates that these tests
-              # failed due to an error, so print the output and fail the
-              # pipeline
-              echo "$output"
-              exit "$status_code"
-            else
-              # The tests must have failed rather than errored out, so
-              # print the output but don't raise an error
-              echo "$output"
-            fi
-          fi
```

Comment:
Now that we're running tests in the `run_iasworld_data_tests` script rather than requiring the script caller to know how to run tests separately, we can reassign ownership over this error logic to the script from the workflow. Ultimately I think this results in simpler code, since the Python utilities for running dbt commands have a clearer set of attributes for distinguishing test failures from test errors (which we'll see shortly).

Reply:
**question (non-blocking):** Is the output still shown in the runner? I assume `dbt.invoke()` shows that output but I don't see a test run.

Reply:
This was a good reminder for me to test a workflow run, which I hadn't done yet (I'd only tested locally) 😅 The `dbt.invoke()` call does indeed print logs in real time, which you can see here: https://github.com/ccao-data/data-architecture/actions/runs/10407263825/job/28822148281#step:5:30

Reply:
See here for a successful workflow run: https://github.com/ccao-data/data-architecture/actions/runs/10407263825/job/28822628961